### DISCUSSION TITLE
Improved ally behavior

### DISCUSSION THREAD

**Discussion by tlsa** [ihhub/fheroes2#9965] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

When playing maps where two or more players are allied, there are various **regressions** compared to the behavior of the original game that are particularly apparent in multi-player:

* **Sharing mage guild spells**
  It used to be possible for an allied player to remove the creatures from a castle and allow their friend to take the castle to learn a good spell. The player could then re-take the castle on the next turn.
  
  This behavior is possible at the moment in fheroes2 only when the players are *not* allied, because allies are prevented from visiting allies castles.
* **Battling with allied heroes**
  At the moment with fheroes2, allies can't battle each others heroes. It used to be fun in the original game, when playing with a friend, at the end of the game to load the last autosave and let both players best heroes fight.

### Describe the possible solution

A good solution for this would be an improved ally mechanic.

* [ ] Allow players to **visit** their allies castles, mines, etc **without taking them**
  * This will allow players to learn the spells at an allied mage guild without the clunky take, re-take mechanic of the original game.
  * If an allied castle contains creatures, the castle is visited without causing a battle.
  * If there is a hero in a castle, it is not possible for an ally to visit without breaking the alliance and fighting the resident hero.
* [ ] Add feature to **break an alliance**
  * There would be a new menu showing alliances and allowing the player to break them.
  * After an alliance is broken, the former allies can take each others mines/castles/etc and fight each others heroes.

**Comment by ihhub** [comment]
This issue correlates with #1321

**Comment by ihhub** [comment]
Moving this item to the discussion section.
